Android Get Started Content Port

zglueck opened the pull request:
This pull request closes #25 in support of the #23 epic.

The content already on https://worldwind.arc.nasa.gov/ for getting started with Android was ported in this pull request. To simplify the content for markdown, some of the expandable boxes and other features were eliminated. Most of the content is word for word from the current site.

The "Get Started" section includes the Gradle instructions for downloading the library.

The CI build is failing because of a bad response from the Bintray API. API requests were added in this pull request in the form of shortcodes to show the latest Android version available from Bintray. I expect that once the Bintray nasaworldwind organization issue is resolved the build will function properly.

zglueck committed:
Remove unnecessary page variable checks

zglueck committed:
Add travis configuration
- See #22 for epic context
- See #12 for complete issue

zglueck committed:
Add homepage
- See #24
- Add supporting images
- Turn on tooltips on the home page
- Add the main page as a partial

zglueck committed:
Add contact page
- See #24

zglueck committed:
Add about page
- See #24

zglueck committed:
Add padding to images

zglueck committed:
Add 404 page

zglueck committed:
Fix branding

zglueck committed:
Fix responsive navbar dropdown

zglueck committed:
Merge develop

zglueck committed:
Update wording
- See discussion with @emxsys in #41

zglueck committed:
Update and augment Android overview

zglueck committed:
Add initial overview content
- See #25

zglueck committed:
Update style and content

Changes are from discussion with @pdavidc and @emxsys on #41.

- Remove second text block from home page header
- Remove tooltips on buttons in home page header
- Remove full page width scaling of header image
- Update images on about page
- Remove extra text in contact page
- Add padding to bottom of the page

zglueck committed:
Merge branch 'feature/#24-common-site-wide-pages' into feature/#25-android-getstarted

zglueck committed:
Adjust jumbotron font

zglueck committed:
Merge branch 'feature/#24-common-site-wide-pages' into feature/#25-android-getstarted

zglueck committed:
Update overview

zglueck committed:
Update get started

zglueck committed:
Merge branch 'develop' into feature/#25-android-getstarted

pdavidc commented:
@zglueck The commit history issue appears in this pull request as well. There's something amiss with the branching and merging here, and I'm concerned that it'll get worse as time goes on. Perhaps rebasing in the feature branch is necessary here:
https://git-scm.com/docs/git-rebase

pdavidc commented on content/android/_index.md:
```diff
@@ -11,4 +11,8 @@ listdescription: ""
 
```
The Overview page seems really sparse. I think we should consolidate it with Get Started, keeping the name Get Started and dropping Overview. The opening paragraphs in both seem like they could be condensed slightly and combined.

zglueck replied:
@pdavidc I'm on board with combining the two pages. The question then becomes what should the "default" page for the projects become, how does its URL look, and how should the project navigation look?

For example, when you go to to `localhost:1313/android/` right now, you get the overview page, which is the "default" page and represented by the "Overview" link in project navigation.

For clarification, "default" has no specific Hugo connotation, I'm speaking in more general permalink/url and site behavior.

I think there a few options. I've outlined them below. Each one addresses the default page, URL pattern, and project navigation changes. I'm leaning towards the first one.

1. "Get Started" becomes the default. The `/android/get-started/` url disappears. Project navigation removes the current "Overview" option and starts with the "Get Started" option. For completeness, we can use Hugo [Aliases](https://gohugo.io/extras/aliases/) to redirect `/android/get-started/` to the default `/android/`.
2. "Get Started" is renamed to something more generic (i.e. Overview, Home, etc.) and the `/android/get-started` URL disappears. The menu option for "Get Started" is replaced with the more generic selection.
3. We come up with more content for the current Overview or implement some other "default" or introduction page.

Actually, as I've been documenting this, I really think the first one is the best option. I'm open to any suggestions or comments, but I'm going to implement the first option until hearing otherwise.

pdavidc replied:
I agree. The first option makes the most sense to me. Thanks for ensuring that the site structure and URLs are sensible.

pdavidc commented on content/android/_index.md:
```diff
@@ -11,4 +11,8 @@ listdescription: ""
+
+WorldWind Android provides high-resolution terrain and imagery, retrieved from remote servers automatically as needed. You can also provide your own terrain and imagery. WorldWind Android additionally provides a rich collection of shapes that you can use to represent information on the globe or in space.
+
+The WorldWind Android client is compatible to [API level](https://source.android.com/source/build-numbers) 19 (KitKat 4.4) and up. It displays the globe via the [GLSurfaceView](https://developer.android.com/reference/android/opengl/GLSurfaceView.html). Read more about how to integrate WorldWind into your app in the [Get Started](/android/get-started/) section. See how to implement basic functiona ...
```
Limit links to those the audience really needs. Over-use of links and formatting makes for a poor reading experience.

Let's drop the references to nonexistent Tutorials and Developer Guide. That way we can actually deploy this site before the entire Android content is complete.

zglueck replied:
The [project navigation bar](https://github.com/NASAWorldWind/NASAWorldWind.github.io/blob/develop/layouts/partials/project-navbar.html) includes links to Tutorials and Downloads. Do you think these should disappear until content is complete?

I'd like to keep the same html template between the projects if possible, but a customized version for each one isn't terribly complex.

pdavidc replied:
If the site is complete as-is then we can publish it sooner rather than later. That's a good thing, and it's what drove me to suggest that we suppress unused links until they point to content. 

With regard to tutorials, I recommended that we split part of the Get Started page into our first tutorials. If we decide to do that, Tutorials would have some content.

Attempting to keep the navigation bar templates consistent across projects makes sense, until we find a need to customize.

pdavidc commented on content/android/get-started.md:
```diff
@@ -0,0 +1,202 @@
+
+WorldWind Android is an [Android Library](https://developer.android.com/studio/projects/android-library.html) which compiles into an Android Archive (AAR) file. The AAR can be linked and used to display the 3D virtual globe within your application. The WorldWind Android AAR is compatible to API level 19 (Android 4.4 KitKat).
+
+Skip to the [**Download**](#download) section for directions on how to add WorldWind Android to your application. If you need some guidance with setting up the Android Studio IDE, the [**Setup**](#setup) section includes in-depth directions and walks through the creation of a sample project utilizing the WorldWind library.
```
I think it's best to omit this description and let the headings to the talking. People don't really read web pages, they scan them. Extra text just makes things look more complex than they really are.

pdavidc commented on content/android/get-started.md:
```diff
@@ -0,0 +1,202 @@
+Skip to the [**Download**](#download) section for directions on how to add WorldWind Android to your application. If you need some guidance with setting up the Android Studio IDE, the [**Setup**](#setup) section includes in-depth directions and walks through the creation of a sample project utilizing the WorldWind library.
+
+---
+### <a name="download"></a>Download
```
What do you think about using the simpler download/JCenter text I used in Android's README.md? I think it explains enough and conveys the necessary information with much less explanation. Of course it would have to be extended to include OJO.

Again, limit links to those the audience really needs. Over-use of links and formatting makes for a poor reading experience.

Limit formatting to only when it's really needed. Over-use of formatting makes pages an eyesore.

pdavidc commented on content/android/get-started.md:
```diff
@@ -0,0 +1,202 @@
+layout: "project"
+---
+
+## Getting Started
```
Page is Get Started but heading is Getting Started.

pdavidc commented on content/android/get-started.md:
````diff
@@ -0,0 +1,202 @@
+```groovy
+compile 'gov.nasa.worldwind.android:worldwind:{{% latestOjoVersion url="https://oss.jfrog.org/artifactory/api/search/versions?g=gov.nasa.worldwind.android&a=worldwind&repos=oss-snapshot-local" %}}'
+```
+---
````
I think these are our first two Tutorials:
- Android Development Environment (or something similar) (lines 43-55)
- Basic WorldWind Android App (lines 56-202)

They'd be much better as tutorials, and would keep the Get Started page simple, with room for minor expansion should we need it.

pdavidc commented on content/android/get-started.md:
```diff
@@ -0,0 +1,202 @@
+#### Install Android Studio
+[Android Studio](https://developer.android.com/studio/index.html) is the IDE used by the WorldWind development team. It includes all the tools you need to build apps for Android. If you haven't done so already, download and install Android Studio.
+
+{{< button url="https://developer.android.com/studio/index.html" buttonText="Get Android Studio" >}}
```
I think we ought to reserve this button image for our own links. It does look good but my first impression was that we (NASAWorldWind) had something to do with installing Android Studio.

pdavidc commented on content/android/get-started.md:
```diff
@@ -0,0 +1,202 @@
+- Read "[Introduction to Android](https://developer.android.com/guide/index.html)" and "[Application Fundamentals](https://developer.android.com/guide/components/fundamentals.html)" in the [Android Developer's Guide](http://developer.android.com/guide/).
+- Create a bookmark to the [Android API Docs](http://developer.android.com/reference/packages.html).
+
+#### How to build a basic WorldWind Android 'app'
```
The layout doesn't seem to work very well in this section. I think refactoring it as a Tutorial would give you room to use a better layout by utilizing headings that you're avoiding now.

Again, be conservative with formatting in the middle of a sentence. We don't need to highlight the name of every code object. It's enough to use correct capitalization. The code blocks however, make perfect sense and look great.

pdavidc commented on layouts/shortcodes/latestBintrayVersion.html:
```diff
@@ -0,0 +1 @@
+{{ $url := .Get "url" }}{{ $versionInfo := getJSON $url }}{{ $versionInfo.name }}
```
I like what you've done here. I wonder if it would be improved by including the group ID and artifact ID as well, thereby making the shortcode specific to Gradle and ensuring that changes to the build artifacts are reflected automatically. For example:

Markdown:
compile '{{% latestGradleConfiguration url="https://api.bintray.com/packages/nasaworldwind/maven/WorldWindAndroid" %}}'

Shortcode:
{{ $url := .Get "url" }}{{ $versionInfo := getJSON $url }}{{ $versionInfo.system_ids[0] }}":"{{ $versionInfo.latest_version }}"@aar"

This shortcode may be incorrect, but I think you'll get the idea. The output is the string:
gov.nasa.worldwind.android:worldwind:0.6.0@aar

pdavidc reviewed (changes requested):
Excellent progress! Your choice of Hugo appears to be paying off. I have the impression that progress is quick and changes are easy, so I gave you my complete list of feedback.

zglueck committed:
Update formatting and content layout

Changes are based on feedback in #42
- Refactor Get Started page content into two tutorials
- Refine download information
- Update images
- Incorporate additional data with Bintray version lookup
- Reduce project navigation bar options until content fills out

pdavidc commented:
Looks good!
